builtin/credential/aws/path_login.go

```diff
@@ -607,8 +607,14 @@ func (b *backend) pathLoginUpdateEc2(ctx context.Context, req *logical.Request,
 	}
 
 	// Check for a CIDR match.
-	if !cidrutil.RemoteAddrIsOk(req.Connection.RemoteAddr, roleEntry.TokenBoundCIDRs) {
-		return nil, logical.ErrPermissionDenied
+	if len(roleEntry.TokenBoundCIDRs) > 0 {
+		if req.Connection == nil {
+			b.Logger().Warn("token bound CIDRs found but no connection information available for validation")
+			return nil, logical.ErrPermissionDenied
```

Comment:
Should this be a 500?

Reply:
Maybe? In practice this should only ever happen in tests so I'm not super concerned, and it logs a warning.